api/python/t4/data_transfer.py

```diff
@@ -594,8 +594,15 @@ def get_bytes(src):
 def get_size_and_meta(src):
     """
     Gets metadata for the object at a given URL.
+
+    If no S3 versionId is present in the URL it will return a versionId for a current object.
```

Comment:
This behavior seems strange to me. I'd expect this method to return the versionid regardless of the inputs. It might also be useful to return the `is_current` value.

Reply:
Yeah, I was trying to avoid parsing the same url twice to check the version id, but i agree that this is convoluted. Switched back to checking in both places.